itests/org.openhab.binding.mqtt.ruuvigateway.tests/src/main/java/org/openhab/binding/mqtt/ruuvigateway/MqttOSGiTest.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mqtt.ruuvigateway;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.openhab.core.config.discovery.inbox.Inbox;
+import org.openhab.core.io.transport.mqtt.MqttBrokerConnection;
+import org.openhab.core.io.transport.mqtt.MqttConnectionState;
+import org.openhab.core.items.ItemProvider;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.items.ManagedItemProvider;
+import org.openhab.core.test.java.JavaOSGiTest;
+import org.openhab.core.thing.ManagedThingProvider;
+import org.openhab.core.thing.ThingProvider;
+import org.openhab.core.thing.link.ItemChannelLinkProvider;
+import org.openhab.core.thing.link.ManagedItemChannelLinkProvider;
+
+import io.moquette.BrokerConstants;
+import io.moquette.broker.Server;
+
+/**
+ * Creates a Moquette MQTT broker instance and a {@link MqttBrokerConnection} for testing MQTT bindings.
+ *
+ * @author Wouter Born - Initial contribution
+ * @author Sami Salonen - Copied to MQTT Ruuvi Gateway addon
+ */
+@NonNullByDefault
+public class MqttOSGiTest extends JavaOSGiTest {
+
+    private static final String BROKER_ID = "test-broker";
+    @SuppressWarnings("null")
+    private static final int BROKER_PORT = Integer.getInteger("mqttbroker.port", 1883);
```

Comment:
Not sure if relevant here, but could there be concurrency issues when running tests in parallel, and more than one test wants to listen on this specific port? See #11523. Perhaps @wborn can advise.

Reply:
It'll work, I see it's based on my own itest code for the homie, homeassistant add-ons. :wink: